src/GovUk.Education.ExploreEducationStatistics.Data.Processor/Models/DataFileReader.cs

```diff
@@ -0,0 +1,425 @@
+        InstitutionName,
+    
+        [EnumLabelValue("new_la_code")]
+        NewLaCode,
+        
+        [EnumLabelValue("old_la_code")]
+        OldLaCode,
+        
+        [EnumLabelValue("la_name")]
+        LaName,
+    
+        [EnumLabelValue("lad_code")]
+        LadCode,
+        
+        [EnumLabelValue("lad_name")]
+        LadName,
+    
+        [EnumLabelValue("local_enterprise_partnership_code")]
+        LepCode,
+        
+        [EnumLabelValue("local_enterprise_partnership_name")]
+        LepName,
+    
+        [EnumLabelValue("mayoral_combined_authority_code")]
+        McaCode,
+        
+        [EnumLabelValue("mayoral_combined_authority_name")]
+        McaName,
+    
+        [EnumLabelValue("trust_id")]
+        MatCode,
+        
+        [EnumLabelValue("trust_name")]
+        MatName,
+    
+        [EnumLabelValue("opportunity_area_code")]
+        OpportunityAreaCode,
+        
+        [EnumLabelValue("opportunity_area_name")]
+        OpportunityAreaName,
+    
+        [EnumLabelValue("pcon_code")]
+        ParliamentaryConstituencyCode,
+        
+        [EnumLabelValue("pcon_name")]
+        ParliamentaryConstituencyName,
+    
+        [EnumLabelValue("provider_ukprn")]
+        ProviderCode,
+        
+        [EnumLabelValue("provider_name")]
+        ProviderName,
+    
+        [EnumLabelValue("region_code")]
+        RegionCode,
+        
+        [EnumLabelValue("region_name")]
+        RegionName,
+    
+        [EnumLabelValue("rsc_region_lead_name")]
+        RscRegionName,
+    
+        [EnumLabelValue("school_urn")]
+        SchoolCode,
+        
+        [EnumLabelValue("school_name")]
+        SchoolName,
+    
+        [EnumLabelValue("sponsor_id")]
+        SponsorCode,
+        
+        [EnumLabelValue("sponsor_name")]
+        SponsorName,
+    
+        [EnumLabelValue("ward_code")]
+        WardCode,
+        
+        [EnumLabelValue("ward_name")]
+        WardName,
+    
+        [EnumLabelValue("planning_area_code")]
+        PlanningAreaCode,
+        
+        [EnumLabelValue("planning_area_name")]
+        PlanningAreaName
+    }
+
+    private static readonly EnumToEnumLabelConverter<TimeIdentifier> TimeIdentifierLookup = new();
+    private static readonly EnumToEnumLabelConverter<GeographicLevel> GeographicLevelLookup = new();
+    
+    private readonly int _timeIdentifierColumnIndex;
+    private readonly int _yearColumnIndex;
+    private readonly int _geographicLevelColumnIndex;
+    private readonly Dictionary<LocationColumn, int> _locationColumnIndexes;
+    
+    private readonly Dictionary<Guid, int>? _indicatorColumnIndexes;
+    private readonly Dictionary<Guid, int>? _filterColumnIndexes;
+    private readonly Dictionary<Guid, int>? _filterGroupColumnIndexes;
+    
+    private readonly Dictionary<string, FilterItem>? _filterItemCache;
+
+    public DataFileReader(
+        List<string> csvHeaders, 
+        SubjectMeta? subjectMeta = null)
```

Comment:
I'm not sure I follow why `subjectMeta` is optional here and why these dictionaries are optional. The methods which rely on them don't handle them being optional.

For example

```
    public Dictionary<Guid, string> GetMeasures(List<string> rowValues)
    {
        return _indicatorColumnIndexes!
            .ToDictionary(
                indicatorMeta => indicatorMeta.Key,
                indicatorMeta => rowValues[indicatorMeta.Value]);
    }
```

Reply:
Hey Ben - yeah, this is because all of the non-optional stuff in DataFileReader is usable both in Stage 2 of the import (before we have the SubjectMeta) and also in Stage 3 (after we have the SubjectMeta).

I could break the class into 2 reader classes, one called "DataFileSubjectMetaReader" and one called "DataFileIndicatorReader" or something along those lines instead?

Reply:
Yeah, I think splitting it up will make it a bit more obvious as to how it's intended to be used and what's required to construct it and when.

It will break up the particularly long constructor of `DataFileReader` here, and remove the optional from these fields, and also have the benefit of removing the null overriding operators throughout the methods using them, so it seems like a good idea to me to split them up.

Reply:
Cool will do!